sdcm/provision/aws/capacity_reservation.py

```diff
@@ -0,0 +1,236 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright (c) 2024 ScyllaDB
+import logging
+from datetime import datetime, timedelta
+from typing import List, Dict, Tuple
+
+from botocore.exceptions import ClientError
+import boto3
+
+LOGGER = logging.getLogger(__name__)
+
+
+# Shlould runner be created in the same az always? - in that case CR needs to be created before sct-runner $0.01/GB data transfer price
```

Comment:
I think we should include SCT

Reply:
in that case, we'll need to split provision step to 2: spin up placement_group and reserve capacity and then create runner then provision step may happen (as need runner_ip to set up syslogng).
Quite complcated - i propose doing it in followup task

Reply:
o.k. please create the task for it.
we'll need to see if we can track the cost

Reply:
https://github.com/scylladb/scylla-cluster-tests/issues/7398